File: DotNet/CesiumLanguageWriter/Advanced/ICesiumArcTypeValuePropertyWriter.cs
namespace CesiumLanguageWriter.Advanced
{
    /// <summary>
    /// A writer that can write a value as a type of an arc.
    /// </summary>
    public interface ICesiumArcTypeValuePropertyWriter : ICesiumPropertyWriter
    {
        /// <summary>
        /// Writes the value expressed as an arc type.
        /// </summary>
        /// <param name="value">The type of an arc.</param>
        void WriteArcType(CesiumArcType value);
    }
}

File: Java/CesiumLanguageWriter/translatedSrc/cesiumlanguagewriter/advanced/ICesiumArcTypeValuePropertyWriter.java
package cesiumlanguagewriter.advanced;


import agi.foundation.compatibility.*;
import cesiumlanguagewriter.*;
import javax.annotation.Nonnull;

/**
 * A writer that can write a value as a type of an arc.
 */
@SuppressWarnings( {
        "unused",
        "deprecation",
        "serial"
})
public interface ICesiumArcTypeValuePropertyWriter extends ICesiumPropertyWriter {
    /**
    * Writes the value expressed as an arc type.
    * @param value The type of an arc.
    */
    void writeArcType(@Nonnull CesiumArcType value);
}